Assets/Enemy/Enemy.cs

```diff
@@ -524,6 +538,19 @@ public class Enemy : MonoBehaviour {
     }
   }
 
+  private IEnumerator HandleTemporaryArmorReduction() {
+    while (true) {
+      yield return new WaitUntil(() => TempArmorReduceEndTime > Time.time);
```

Comment:
This wont handle a new armor tear happening.  It will be waiting for the last one to conclude.

Reply:
This was a great point, I've adjusted the armor handling into the temporarily reduce armor method.

Reply:
Still a bit of a bug, weird frame stuff but its possible we never reset armor.